apache/trafficcontrol
# GitHub Actions ATS RPM: Verify that the cached or built RPM matches the expected RPM name

This PR adds validation to ensure that the GitHub Actions cache key corresponds with the name of the ATS RPM that that entry contains. This fixes an issue that currently makes the *CDN in a Box CI* workflow fail, because the cached RPM name is not what CDN in a Box expects the RPM name to be.

Before:
Cache key: `965df952e1da3e5039473db8cf219cf9ef285a93-8.1.4-el8-aebdb313c2d32e7a83d8aec597b9d91e0778f2fa9087f35fee28632781586b98`
Cached RPM: `trafficserver-8.1.3-10.965df952e.el8.x86_64.rpm`
CiaB error:
> ```shell
> cp: cannot stat '../../dist/trafficserver-8.1.4-0.965df952e.el8.x86_64.rpm': No such file or directory
> ```

As of this PR:
Cached RPM: `trafficserver-8.1.4-0.965df952e.el8.x86_64.rpm`
Cache key: `trafficserver-8.1.4-0.965df952e.el8.x86_64.rpm-aebdb313c2d32e7a83d8aec597b9d91e0778f2fa9087f35fee28632781586b98
`
No CiaB error
<hr/>

## Which Traffic Control components are affected by this PR?

- Automation - building the Apache Traffic Server RPM
- unknown

## What is the best way to verify this PR?

- Verify that the *Traffic Control Cache Config integration tests* GHA workflow passes
- Verify that the *CDN-in-a-Box CI* GHA workflow passes
- Verify that the *Traffic Control Health Client integration tests* GHA workflow passes

## PR submission checklist
